### ISSUE TITLE
Dealing with S3 driver private files for FileUpload Field & Image Column 

### ISSUE THREAD

**Issue by tomheno** [laravel-filament/filament#156]
**Describe the bug**
I was not able to make the [FileUpload **form field**](https://filamentadmin.com/docs/forms/#fields-file-upload) (1 image) Field and [Image **table column**](https://filamentadmin.com/docs/tables/#columns-image) (the same image) work properly with a **private** S3 filesystem driver, with **private** file visibility. 

**To reproduce**
1. Create a form with the following field : `FileUpload::make('uri')->disk('s3')->directory('storage')->visibility('private')`
2. Go to this form & upload a photo, this will work, the temporary livewire upload pre-submittion and the final upload on form submittion will proceed.
3. It'll redirect you to the entry "edit" page, where the image wont load, with the following network message from S3 (error code 403) : 
`<Error><Code>AccessDenied</Code><Message>Access Denied</Message><RequestId>...</RequestId><HostId>...</HostId></Error>`
4. It's the same problem with the Image Column so I won't detail unless it's necessary
5. What is happening is that the Field is trying to load the image from a **public** url, without using the built-in Laravel S3 driver feature to request a **temporary public URL** out of a private file on S3.

**Expected behavior**
The image should load as expected from S3, once it's privately stored.

**Way to fix**
I managed to get everything working properly by tweaking a little the forms/src/HasForm **getUploadedFileUrl** behavior, and the src/Columns/Image **getPath** method, but it need some improvement to be a PR, as I don't know how to make it conditionnal for the HasForm **getUploadedFileUrl** method (I made a **private** parameter for the Image Column).
You can check my commit here : [https://github.com/tomheno/filament-with-s3-private-file-handling/commit/ddb0cd3f60e4c740c7349599b3e0f2918a9b9247](https://github.com/tomheno/filament-with-s3-private-file-handling/commit/ddb0cd3f60e4c740c7349599b3e0f2918a9b9247).

**Context**
- Package version: 1.4.5
- Laravel version: 8.12
- Browser: Chrome
- Server OS: Windows
- PHP version: 8

**Additional details**
Thanks in advance for your time and consideration on this issue ! 😀

**Comment by ryangjchandler** [comment]
Closed by #157.

**Comment by Xoshbin** [comment]
Hi,
I'm having the same issue, I can upload the file to the server the same way as @tomheno , but it's not loading or showing the image in the table or in the form.
that's my environment and the form.
`FILESYSTEM_DRIVER=local
`when I'm changing the above in variable in .env file, at upload the image is just loading without uploading it to the s3.
```
Components\FileUpload::make('attachment')->label('Attachments')
                            ->disk('s3')
                            ->directory('storage')
                            ->loadingIndicatorPosition($position = 'right') // Set the position of the loading indicator.
                            ->removeUploadButtonPosition($position = 'left') // Set the position of the remove upload button.
                            ->uploadButtonPosition($position = 'right') // Set the position of the upload button.
                            ->uploadProgressIndicatorPosition($position = 'right') // Set the position of the upload progress indicator.
                            ->imagePreviewHeight(400)
                            ->visibility($visibility = 'private'),
```
